lib/docker/executeDockerBuild.ts

```diff
@@ -315,16 +331,67 @@ async function pushEnabled(gi: ProjectAwareGoalInvocation, options: DockerOption
     // tslint:disable-next-line:no-boolean-literal-compare
     if (options.push === true || options.push === false) {
         push = options.push;
-    } else if ((!!options.user && !!options.password) || !!options.config) {
+    } else if (options.registries.some(r => !!(r.username && r.password)) || !!options.config) {
```

Comment:
Likely to fail in cases where registry is unset.

Reply:
👍 Will set a default [] for registries as discussed